config/pmd_controversial.xml

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Catroid: An on-device visual programming system for Android devices
+  ~ Copyright (C) 2010-2015 The Catrobat Team
+  ~ (<http://developer.catrobat.org/credits>)
+  ~
+  ~ This program is free software: you can redistribute it and/or modify
+  ~ it under the terms of the GNU Affero General Public License as
+  ~ published by the Free Software Foundation, either version 3 of the
+  ~ License, or (at your option) any later version.
+  ~
+  ~ An additional term exception under section 7 of the GNU Affero
+  ~ General Public License, version 3, is available at
+  ~ http://developer.catrobat.org/license_additional_term
+  ~
+  ~ This program is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  ~ GNU Affero General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Affero General Public License
+  ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    name="Catroid PMD unnecessary ruleset"
+    xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+    xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd" >
+    <description >PMD Java Controversial Ruleset for Catroid/PocketCode</description >
+    <rule ref="rulesets/java/controversial.xml" >
+        <exclude name="AssignmentInOperand" />
+        <exclude name="AtLeastOneConstructor" />
+        <exclude name="AvoidFinalLocalVariable" />
+        <exclude name="AvoidPrefixingMethodParameters" />
+        <exclude name="AvoidLiteralsInIfCondition" />
+        <exclude name="BooleanInversion" />
+        <exclude name="CallSuperInConstructor" />
+        <exclude name="DataflowAnomalyAnalysis" />
+        <exclude name="NullAssignment" />
+        <exclude name="OnlyOneReturn" />
+        <exclude name="UseConcurrentHashMap" />
+        <exclude name="UseObjectForClearerAPI" />
+        <exclude name="AvoidUsingShortType" />
+        <exclude name="DefaultPackage" />
+        <exclude name="UnnecessaryConstructor" />
+        <exclude name="DoNotCallGarbageCollectionExplicitly">
+            <exclude-pattern>.*src\org\catrobat\catroid\ui\BaseActivity.java</exclude-pattern>
+        </exclude>
+        <exclude name="AvoidUsingVolatile">
+            <exclude-pattern>.*src\org\catrobat\catroid\content\Script.java</exclude-pattern>
```

Comment:
[This](https://github.com/Catrobat/Catroid/blob/e0af7413aee3f28f4cb1cd210b48ea998934ec71/catroid/src/org/catrobat/catroid/content/Script.java#L51) is the only volatile variable which in the meantime is a relic of the old thread-based handling of actions.
The volatile modifier has its use-cases and this rule should be excluded completely.
BUT I'm not familiar with the Android asynch tasks/jobs/etc but they may also rely on the volatile modifier, I guess?